src/wallet/bdb.cpp

```diff
@@ -58,12 +58,12 @@ bool WalletDatabaseFileId::operator==(const WalletDatabaseFileId& rhs) const
  * erases the weak pointer from the g_dbenvs map.
  * @post A new BerkeleyEnvironment weak pointer is inserted into g_dbenvs if the directory path key was not already in the map.
  */
-std::shared_ptr<BerkeleyEnvironment> GetBerkeleyEnv(const fs::path& env_directory)
+std::shared_ptr<BerkeleyEnvironment> GetBerkeleyEnv(const fs::path& env_directory, bool use_shared_memory)
```

Comment:
What's the rationale for using "shared memory" here instead of just "private"? Especially as it comes from a "-privdb" flag and is eventually converted to a `DB_PRIVATE` flag.
Another option would be to use `std::option<fs::path>` and a single argument. After all, the path is irrelevant in case of a private database.

Reply:
I've never heard of a "private database" before this PR and I think that terminology is a little arcane, but shared memory is a concept I'm familiar with, and that's what the flag controls, so that's what i called the DatabaseOptions member. This particular function argument and a few local variables are named to be consistent with the DatabaseOptions field name, but if you want to call them something different, that sounds fine. Maybe say what specific variables you would like to rename and what you would like to call them.

Reply:
https://github.com/observerdev/db-4.8.30/blob/3c75267f11336acb1be06bed419e8056cc689d09/dbinc/region.h#L26-L28 - Is this possibly the correct explanation of `DB_PRIVATE`?

Reply:
> https://github.com/observerdev/db-4.8.30/blob/3c75267f11336acb1be06bed419e8056cc689d09/dbinc/region.h#L26-L28 - Is this possibly the correct explanation of `DB_PRIVATE`?

I'm sure it's correct. I don't think is any question about what that flag does. The questions here are just about readability and variable names in this PR. I don't think there's anything that needs to be done yet, except rebase

Reply:
That was supposed to be for my information :)